bitnami/geode/templates/locator/statefulset.yaml

```diff
@@ -0,0 +1,385 @@
+            {{- end }}
+          {{- end }}
+          {{- if .Values.locator.resources }}
+          resources: {{- toYaml .Values.locator.resources | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: locator
+              containerPort: {{ .Values.locator.containerPorts.locator }}
+            - name: http
+              containerPort: {{ .Values.locator.containerPorts.http }}
+            - name: rmi
+              containerPort: {{ .Values.locator.containerPorts.rmi }}
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.locator.livenessProbe.enabled }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.locator.livenessProbe "enabled") "context" $) | nindent 12 }}
+            exec:
+              command:
+                - /bin/bash
+                - -ec
+                - |
+                  gfsh status locator --dir /bitnami/geode/data >/dev/null 2>&1 || exit 1
+          {{- else if .Values.locator.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.locator.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.locator.readinessProbe.enabled }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.locator.readinessProbe "enabled") "context" $) | nindent 12 }}
+            exec:
+              command:
+                - /bin/bash
+                - -ec
+                - |
+                  gfsh status locator --dir /bitnami/geode/data >/dev/null 2>&1 || exit 1
+          {{- else if .Values.locator.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.locator.customReadinessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.locator.startupProbe.enabled }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.locator.startupProbe "enabled") "context" $) | nindent 12 }}
+            tcpSocket:
+              port: locator
+          {{- else if .Values.locator.customStartupProbe }}
+          startupProbe: {{- include "common.tplvalues.render" (dict "value" .Values.locator.customStartupProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.locator.lifecycleHooks }}
+          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.locator.lifecycleHooks "context" $) | nindent 12 }}
+          {{- end }}
+          {{- end }}
+          volumeMounts:
+            - name: data
+              mountPath: /bitnami/geode
+            {{- if or .Values.locator.configuration .Values.locator.existingConfigmap }}
+            - name: configuration
+              mountPath: /bitnami/geode/config/gemfire.properties
+              subPath: gemfire.properties
```

Comment:
Any reason why we want to force a subPath when we could just mount the whole configmap? I can see useful for users that want to mount other config files (not sure if it is applicable in this application, but in solutions like Cassandra it is very useful)

Reply:
We have specific configmaps for specific config files (see `configuration` and `log4j` parameters). Therefore, we can specify the subpath.